### ISSUE TITLE
[DateRangePicker] Clicking the second input clear the first input

### ISSUE THREAD

**Issue by marc-polizzi** [mui/mui-x#4314]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

- The date range picker is empty.
- I enter a value in the "from" by clicking the from and using the keyboard.
- Then I click on the "to" and the "from" field is being cleared out.

### Expected behavior 🤔

The "to" should not be cleared out. Same behavior as using the "tab" key to switch from the "from" field to the "to" field.

### Steps to reproduce 🕹

Steps:

1. you can try the Basic Usage : https://mui.com/components/date-range-picker/#basic-usage
2.
3.
4.

### Your environment 🌎

Chrome 99 (windows + linux) Firefox 98 (linux)

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by flaviendelangle** [comment]
Hi,

The pickers are being migrated to the X repository
The new packages will most likely be published next week
I will have a look to fix this problem in the new package :+1:

**Comment by cherniavskii** [comment]
I can reproduce the issue here https://mui.com/x/react-date-pickers/date-range-picker/#basic-usage

https://user-images.githubusercontent.com/13808724/163834247-87022af1-7036-4b69-80fa-7c4fdc1d1f62.mov

**Comment by flaviendelangle** [comment]
Related to #4439, the closing on blur / opening on focus behavior is not correctly implemented

**Comment by flaviendelangle** [comment]
Fixed by #4408
